### PR TITLE
✨ Introduce doc vars for releases

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR introduces some new documentation variables that align with our current release concepts of "latest regular release" and "latest release". This replaces several self-references with just one (the definition of `ks_latest_release`).

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
Preview at https://mikespreitzer.github.io/kcp-edge-mc/doc-release-vars/

**Comment by ezrasilvera** [comment]
@MikeSpreitzer I'm a little bothered with the "latest-regular-release"  Vs "latest-release" . Why do we need references for  release candidates ?   It also implies that the "latest-release" is somehow "irregular" or "special" ?  
Maybe we should adopt the "stable"  ?  For us, we always said that for now regular==stable (as we don't have stable tests for "stable" )  - I think that having "latest" Vs "latest-stable"  is better

**Comment by eaepstein** [comment]
I thought KS already has two release types: official releases and RC releases, and latest simply points at the most recent official release. Is that no longer the case? Is latest-regular another name for "latest non-RC release"?

**Comment by MikeSpreitzer** [comment]
@ezrasilvera: @eaepstein guessed the answer. Because candidate releases (RCs) are releases, we have established the term "regular" release to distinguish a release that is not an RC from one that _is_ an RC (or alpha or anything else irregular). Thus: "latest regular release" is the latest release that is not an RC (or alpha or...), while "latest release" is the latest release of any sort (including RC or alpha or ...). This is the terminology that we discussed earlier and have been using.

The self references need to refer to _self_ --- even when that self is an RC or alpha --- so the most commonly used variable is `ks_latest_release`.

**Comment by eaepstein** [comment]
What is the difference between RC and alpha releases? May there could just be "releases" and RC-releases", with "latest" pointing at the most recent "release"?

**Comment by MikeSpreitzer** [comment]
"alpha" is something that @francostellari cajoled me into doing. It is a development snapshot that falls short of the goals for the next regular release. @francostellari's new Helm chart can only use a release, so we needed a release of some sort in order to be able to test and continue developing the Helm chart.

A candidate release is one that, as far as we know, meets the goals for the next regular release.

It will be linguistically confusing if we say that an "RC-release" is _not_ a "release".

Do we really need to re-litigate the terminology right now, as we are trying to get the next regular release out?

**Comment by MikeSpreitzer** [comment]
@eaepstein: The `mike` alias named "latest" points to the latest regular release. And the `mike` version warning banner says "latest release" where it means "latest _regular_ release". This is admittedly inconsistent with the rest of our terminology. This is the result of our historical development path and general hurry.

**Comment by eaepstein** [comment]
>Do we really need to re-litigate the terminology right now, as we are trying to get the next regular release out?
Seems like "alpha" and "regular" are terms introduced without discussion.

**Comment by MikeSpreitzer** [comment]
@eaepstein: I definitely remember discussing "regular". The "alpha" was something that we had earlier, then agreed that we do not need any more, then was re-introduced in a rush with only a few people involved.

**Comment by ezrasilvera** [comment]
I believe we agreed that Latest should always point to a "regular" release.   there is no such thing as "latest RC" or "latest Alpha"

**Comment by MikeSpreitzer** [comment]
@ezrasilvera: you may be referring to the `mike` alias named "latest". We have a terminology snafu here. The `mike` alias "latest" points to the "latest regular release", as you observed. If we were completely consistent then we would rename that `mike` alias from "latest" to "latest_regular". The name of this alias was established long ago, before our current thinking about releases, and nobody has bothered to change it. Also, the `mike` banner warning about version is also outdated, speaking of "latest release" when it means "latest regular release".

I really do not think we want to have to tell people that "a candidate release is not a release". We needed an explicit word to distinguish candidate releases from non-candidate releases. That is why we agreed to call the non-candidate releases "regular". This was when we expected no "alpha" releases. Now that we also have had some "alpha" releases, this still does not change the need for an explicit word to distinguish the regular releases from the other kinds of releases.

The mkdocs variables that we define are not connected to `mike` aliases. I chose to introduce variables consistent with our latest thinking and terminology. If anything, I think that the outdated stuff in `mike` should be updated.

**Comment by MikeSpreitzer** [comment]
In plain English, since we have candidate releases, one of them is always the latest. So we _do_ have a latest candidate release. Same for alpha. In plain English, the latest release would be the latest release of any kind.

**Comment by MikeSpreitzer** [comment]
The force-push to c1c306125 is a rebase onto `main`.

**Comment by MikeSpreitzer** [comment]
The force-push to 533216965 is a rebase onto `main`.

File: docs/content/direct/release.md [hunk 1]
```diff
@@ -38,16 +38,12 @@ Making a new kubestellar release requires a contributor to do the following thin
 
 - If not already in effect, declare a code freeze. There should be nothing but bug fixes and doc improvements while working towards a regular release.
 
-- Edit the source for the KCM PCH (in `config/postcreate-hooks/kubestellar.yaml`) and update the tag in the reference to the KCM container image (it appears in the last object, a `Job`).
+- Edit `docs/mkdocs.yml` and update the definition of `ks_latest_release` to `$version` (e.g., `'0.23.0-rc42'`). If this is a regular release then also update the definition of `ks_latest_regular_release`.
```
Comment (ezrasilvera) [inline docs/content/direct/release.md:41]: As we discussed elsewhere - I think we have a notion of "latest" only for regular releases.

File: docs/README.md [hunk 2]
```diff
@@ -101,6 +101,8 @@ There are many global variables defined in the <a href="{{ config.repo_raw_url }
     - edit_uri: {{ config.edit_uri }}
     - ks_branch: {{ config.ks_branch }}
     - ks_tag: {{ config.ks_tag }}
+    - ks_latest_regular_release: {{ config.ks_latest_regular_release }}
+    - ks_latest_release: {{ config.ks_latest_release }}
```
Comment (ezrasilvera) [inline docs/README.md:105]: I suggest we converge to only use one latest (of a regular release) .   All references except one are using the latest of regular release in any case.

Comment (MikeSpreitzer) [reply docs/README.md:105]: No, most of the uses are for the self-references that need to be to the release being made regardless of what kind of release it is.

**Comment by KPRoche** [comment]
Just a note that the "latest release" message is in the main.html file and therefore easily edited:
(this is from [kproche.github.io/](https://kproche.github.io/kubestellar/doc-new-v2b/) )
image

File: docs/content/direct/release.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/direct/release.md:41]: To move the discussion into a public forum, created a GitHub "discussion". See https://github.com/kubestellar/kubestellar/discussions/2218 .

File: docs/README.md (hunk 2 above)
Comment (ezrasilvera) [reply docs/README.md:105]: @MikeSpreitzer (@eaepstein @pdettori ) I don't agree.  I think that at this stage of the project (it may be changed in the future) we want users to use only "regular release"  . The other releases are point releases aimed for developers for specific task (test a fix, test a new feature, etc..) 
As our web-site is aimed to users - all references to 1latest1 on the web site should refer to the latest `regular release`.   
The few developers who use/test the RCs don't need to use direct references for theses RCs on our web documentation in order to use it. 
BTW, I believe that most of the team actually supported to even  generate the docs only for `regular releases`  , and not for other intermediate releases, but we didn't vote on that and just let it continue :(

Comment (eaepstein) [reply docs/README.md:105]: >The other releases are point releases aimed for developers for specific task (test a fix, test a new feature, etc..)

Does this translate to mean 0.23.0 is a "regular" release but 0.23.1 is not? And therefor 0.23.1 cannot be labeled as "latest" ?

Comment (MikeSpreitzer) [reply docs/README.md:105]: @ezrasilvera:

> @MikeSpreitzer (@eaepstein @pdettori ) I don't agree. I think that at this stage of the project (it may be changed in the future) we want users to use only "regular release" .

I agree with that.

The problem is all the self-references that have to be updated when making a release of any kind, including a test (candidate) release. See https://github.com/kubestellar/kubestellar/blob/v0.23.0/docs/content/direct/release.md#making-a-new-kubestellar-release . Most of those self-references are in doc files. I introduced `config.ks_latest_release` to be used in place of those self-references, so that making a release of any kind (including a test release) requires editing only the definition of `config.ks_latest_release` instead of all the places where it appears in the docs.

Comment (MikeSpreitzer) [reply docs/README.md:105]: @eaepstein: I think that when @ezrasilvera wrote about "point releases" he did not mean "point" in the technical semver sense of major.minor.patch; rather, he meant "point" in the sense of "intended for a narrow purpose". A release identified simply as "major.minor.patch" _is_ a regular release, not a test (candidate) release or alpha or beta or other unusual thing.

Comment (MikeSpreitzer) [reply docs/README.md:105]: @ezrasilvera: can we separate a change to our policy of which releases appear on the website from this PR?

**Comment by ezrasilvera** [comment]
@MikeSpreitzer Let's merge this and continue the discussion elsewhere 
/lgtm
/approve
